app/services/email.py

```
@@ -0,0 +1,221 @@
+        field_name: str,
+    ) -> list[str] | None:
+        if not raw_addresses:
+            return []
+        try:
+            return json.loads(raw_addresses)
+        except (json.JSONDecodeError, TypeError):
+            logger.error(
+                "Invalid %s JSON for email %s while checking idempotency payload",
+                field_name,
+                email_id,
+            )
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)